zcash_client_sqlite/src/wallet.rs

```diff
@@ -588,6 +690,74 @@ pub fn get_nullifiers<P>(
     Ok(res)
 }
 
+/// Returns the nullifiers for the notes that this wallet is tracking.
+pub fn get_all_nullifiers<P>(
+    wdb: &WalletDb<P>,
+) -> Result<Vec<(AccountId, Nullifier)>, SqliteClientError> {
+    // Get the nullifiers for the notes we are tracking
+    let mut stmt_fetch_nullifiers = wdb.conn.prepare(
+        "SELECT rn.id_note, rn.account, rn.nf
+            FROM received_notes rn",
+    )?;
+    let nullifiers = stmt_fetch_nullifiers.query_map(NO_PARAMS, |row| {
+        let account = AccountId(row.get(1)?);
+        let nf_bytes: Vec<u8> = row.get(2)?;
+        Ok((account, Nullifier::from_slice(&nf_bytes).unwrap()))
+    })?;
+
+    let res: Vec<_> = nullifiers.collect::<Result<_, _>>()?;
+    Ok(res)
+}
+
+/// Returns unspent transparent outputs that have been received by this wallet at the given
+/// transparent address, such that the block that included the transaction was mined at a
+/// height less than or equal to the provided `max_height`.
+#[cfg(feature = "transparent-inputs")]
+pub fn get_unspent_transparent_outputs<P: consensus::Parameters>(
+    wdb: &WalletDb<P>,
+    address: &TransparentAddress,
+    max_height: BlockHeight,
+) -> Result<Vec<WalletTransparentOutput>, SqliteClientError> {
+    let mut stmt_blocks = wdb.conn.prepare(
+        "SELECT u.prevout_txid, u.prevout_idx, u.script, u.value_zat, u.height, tx.block as block
+         FROM utxos u
+         LEFT OUTER JOIN transactions tx
+         ON tx.id_tx = u.spent_in_tx
+         WHERE u.address = ?
+         AND u.height <= ?
+         AND block IS NULL",
```

Comment:
If the `utxos` table were altered to have a connection to the containing transaction, this would be clearer as:
```suggestion
         AND spent_block IS NULL",
```

Reply:
This should be addressed after https://github.com/zcash/librustzcash/issues/369

Reply:
Opened https://github.com/zcash/librustzcash/issues/503 for this.